spiped/dispatch.c

```diff
@@ -250,6 +302,16 @@ void
 dispatch_shutdown(void * dispatch_cookie)
 {
 	struct accept_state * A = dispatch_cookie;
+	struct conn_list_node * ptr;
+	struct conn_list_node * ptr_next;
+
+	/* Shutdown any open connections. */
+	ptr = A->conn_cookies;
+	while (ptr != NULL) {
+		ptr_next = ptr->next;
+		proto_conn_drop(ptr->conn_cookie);
+		ptr = ptr_next;
```

Comment:
Oh... hmm.  Yeah, we definitely need a comment to that effect.  Or possibly change `proto_conn_drop` to something like `proto_conn_free` which cleans up but doesn't perform the callback -- I've learned that it's possible to tie oneself in knots very easily by having "cancel"/"shutdown" functions make callbacks -- but in this case I don't think that's necessary.  (The scenario we absolutely must avoid is having proto_conn_drop call back to callback_conndied and then callback_conndied makes more calls to proto_conn_drop, since that could give us stack-overflowing recursion; but I'm 99% certain that we're not doing anything like that here.)

Also, I think that means this code can be simplified, to something along the lines of
`while (A->conn_cookies != NULL) proto_conn_drop(A->conn_cookies->conn_cookie);` ?